climada_petals/hazard/wildfire.py

```diff
@@ -1329,3 +1888,209 @@ def _fill_intensity_max(num_centr, ind, index_uni, lat_lon_cpy, fir_bright):
             brightness_ev[0, ind[idx]] = max(brightness_ev[0, ind[idx]], \
                          np.max(fir_bright[lat_lon_cpy == index_uni[idx]]))
     return brightness_ev
+
+def from_sentinel(filename):
+    """Read ids of contained bands and month covered by Sentinel data
+
+    Parameters
+    ----------
+    input_dir : Path or str
+        path to input data directory,
+    filename : string
+        name of netcdf file in input_dir.
+
+    Returns
+    ----------
+    id_bands: list
+        list of indices of the bands contained in the input file
+    event_list:list
+        list of events contained in the input file in the format 'yyyy-mm-dd'
+
+    """
+
+    (_, _, _, date, _, _, _) = filename.split('_')
+    year = date[0:4]
+    month = date[4:6]
+    day = date[6:8]
+    event_list = [year + '-' + month + '-' + day]
+    id_bands = [1]
+
+    return id_bands, event_list
+
+def from_firemip(input_dir, filename):
+    """Read ids of contained bands and month covered by Firemip data
+
+    Parameters
+    ----------
+    input_dir : Path or str
+        path to input data directory,
+    filename : string
+        name of netcdf file in input_dir
+
+    Returns
+    ----------
+    id_bands: list
+        list of indices of the bands contained in the input file
+    event_list:list
+        list of events contained in the input file in the format 'yyyy-mm-dd'
+
+    """
+
+    # read indixes of bands to be extracted
+    raster = rasterio.open(str(Path(input_dir, filename)))
+    nr_bands = raster.count
+    id_bands = np.arange(1, nr_bands+1).tolist()
+    (_, _, _, _, _, _, _, _, _, startyear, endyearnc) = filename.split('_')
+    endyear, _ = endyearnc.split('.')
+    event_names = [str(n) for n in range(int(startyear), int(endyear) + 1)]
+    event_list = [event_ + '-01-01' for event_ in event_names]
+
+    return id_bands, event_list
+
+def firemip_dowscaling(haz_firemip, haz_prob):
+    """Return wildfire hazard consisting of wildfire seasons that were assigned to
+    FireMIP pixel with a similar value of burnt area per year / fire season
+
+    Parameters
+    ----------
+    haz_firemip : WildFire
+        WildFire hazard instance with fraction of burnt area as intensity
+    haz_prob : WildFire
+        Probabilistic WildFire hazard
```

Comment:
These are very confusing names...

Reply:
Do you have a suggestion for better variable names?